File: board_dsl/src/main/java/com/dooho/board/controller/UserController.java
package com.dooho.board.controller;

import com.dooho.board.dto.*;
import com.dooho.board.dto.user.MyPageDto;
import com.dooho.board.dto.user.PatchUserDto;
import com.dooho.board.dto.user.PatchUserResponseDto;
import com.dooho.board.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/user")
public class UserController {

    private final UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @CrossOrigin(origins = "*")
    @PatchMapping("/edit")
    public ResponseDto<PatchUserResponseDto> patchUser(@RequestBody PatchUserDto requestBody, @AuthenticationPrincipal String userEmail){
        ResponseDto<PatchUserResponseDto> result = userService.patchUser(requestBody,userEmail);

        return result;
    }

    @GetMapping("/myPage")
    public ResponseDto<MyPageDto> myPage(@AuthenticationPrincipal String userEmail){
        ResponseDto<MyPageDto> result = userService.myPage(userEmail);

        return result;

    }
}

File: board_spring/src/main/java/com/dooho/board/controller/UserController.kt
package com.dooho.board.controller

import com.dooho.board.dto.ResponseDto
import com.dooho.board.dto.user.MyPageDto
import com.dooho.board.dto.user.PatchUserDto
import com.dooho.board.dto.user.PatchUserResponseDto
import com.dooho.board.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/user")
class UserController @Autowired constructor(private val userService: UserService) {
    @CrossOrigin(origins = ["*"])
    @PatchMapping("/edit")
    fun patchUser(
        @RequestBody requestBody: PatchUserDto?,
        @AuthenticationPrincipal userEmail: String?
    ): ResponseDto<PatchUserResponseDto?> {
        return userService.patchUser(requestBody, userEmail)
    }

    @GetMapping("/myPage")
    fun myPage(@AuthenticationPrincipal userEmail: String?): ResponseDto<MyPageDto?> {
        return userService.myPage(userEmail)
    }
}